phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCMiscIT.java

```diff
@@ -416,6 +706,116 @@ public void testSelectTimeRangeQueries() throws Exception {
         }
     }
 
+    @Test
+    public void testSelectCDCRebuildIndex() throws Exception {
+        Properties props = new Properties();
+        props.put(QueryServices.TASK_HANDLING_INTERVAL_MS_ATTRIB, Long.toString(Long.MAX_VALUE));
+        props.put("hbase.client.scanner.timeout.period", "6000000");
+        props.put("phoenix.query.timeoutMs", "6000000");
+        props.put("zookeeper.session.timeout", "6000000");
+        props.put("hbase.rpc.timeout", "6000000");
+        Connection conn = DriverManager.getConnection(getUrl(), props);
+        String tableName = generateUniqueName();
+        conn.createStatement().execute(
+                "CREATE TABLE  " + tableName + " ( k INTEGER PRIMARY KEY," + " v1 INTEGER, v2 INTEGER, v3 INTEGER, v4 INTEGER)");
+        String cdcName = generateUniqueName();
+
+        conn.createStatement().execute("UPSERT INTO " + tableName + " (k, v1, v2) VALUES (1, 100, 1000)");
```

Comment:
Let's have more row versions here and retrieve these versions with multiple select statements with different time ranges, that is the where clauses should include the corresponding PHOENIX_ROW_TIMESTAMP() ranges.